### DISCUSSION TITLE
Importing Power Platform Environments, terraform apply

### DISCUSSION THREAD

**Discussion by ronmanthe** [microsoft/terraform-provider-power-platform#500] [Q&A]
I am currently working on importing all of our Power Platform resources. I imported my first environment, and the only change to the environment is change to the description of environment in my tfvars file. 

For some reason, when I do a terraform apply, the environment URL as well as many other attributes come back as "(known after apply)" even though these values are populated in the state file. Then when I go through with the apply, it fails because it is attempting to update the URL to the current value:

![image](https://github.com/user-attachments/assets/3f48395d-68e1-4388-86dd-0b1d1408b665)

![image](https://github.com/user-attachments/assets/09911a00-5430-4a34-83aa-2a91788af6cd)

When I change the description in my tfvars file to what it is currently, a terraform plan returns no changes. 

On a side note, I wish we could import power platform users.

**Comment by mawasile** [comment]
Hi @ronmanthe 

Are you using newest 3.1.0 provider version? 
Is your domain a custom one that you defined? 
In your first print screen, the description does not match the actual value.

**Reply by ronmanthe** [reply]
Yes we are currently using 3.1.0.
This is a custom domain. 
Yes, we are trying to update the description.

**Comment by mawasile** [comment]
@ronmanthe I was able to reproduce your issue: #502 The fix will come in the next release.

**Reply by ronmanthe** [reply]
@mawasile  Thank you! Also, is there a reason these values are saying (known after apply)?

**Reply by mawasile** [reply]
"know after apply" is shown by attributes that could change on the cloud side between terraform runs, and that new value would be accepted as a correct (new) one.

**Reply by sam-bryant** [reply]
@mawasile - I'm not following your comment on "known after apply".  These items are all known, exist in the state file, and the provider should know what it is setting the values to.  We feel the known after apply items shouldn't even be showing up because the provider should know what they are.

They do NOT show up when @ronmanthe  isn't changing the description, so in that scenario it appears to know that there aren't any changes.  They only show up when we are actually changing an attribute.  This behavior seems off to us.

**Reply by mawasile** [reply]
@sam-bryant fair point, let me look at it closer, it's possible that some of them where left to re-computer every time and they shouldn't be. Thanks for spotting.

**Reply by mawasile** [reply]
fixed for environment creation/import are in main branch and will get be available with next release after the current 3.2.0
